src/requirements/requirement-frontend-utils.ts

```diff
@@ -247,15 +251,40 @@ type MatchedRequirementFulfillmentSpecification =
 
 /**
  * The function respects the user choice on toggleable requirement, and provides the already decided
- * fulfillment strategy to follow.
+ * fulfillment strategy to follow. It also filters eligible courses based on the user's college/major.
  *
  * @returns a spec telling how the requirement progress should be computed, or null if the requirement
  * is self-check.
  */
 export function getMatchedRequirementFulfillmentSpecification(
   requirement: RequirementWithIDSourceType,
+  { college }: AppOnboardingData,
   toggleableRequirementChoices: AppToggleableRequirementChoices
 ): MatchedRequirementFulfillmentSpecification {
+  const { sourceType, sourceSpecificName } = requirement;
+  const filterEligibleCoursesByRequirementConditions = (
+    coursesList: readonly (readonly number[])[],
+    conditions: RequirementCourseConditions | undefined
+  ) =>
+    coursesList.map(courses =>
+      courses.filter(courseId => {
+        // no requirement conditions
+        if (!(conditions && courseId in conditions)) return true;
+        const { colleges, majorsExcluded } = conditions[courseId];
+        // user college is in colleges list
+        if (college && !colleges.includes(college)) return false;
+        // requirement is in colleges list
```

Comment:
Could you give an example to test where a course would not fulfill a requirement because the college was not in the colleges list? Struggling to come up with one.

Reply:
This is anywhere there is an empty list for a particular college (see ExamCredit.ts). For example, AP Math BC (Engineering) should not count unless the user's colleges includes Engineering. Otherwise, the user should use the Non-Engineering AP course.